modules/tensor_mechanics/src/materials/Compute2DFiniteStrain.C

```diff
@@ -74,17 +74,15 @@ Compute2DFiniteStrain::computeProperties()
     ave_Fhat /= _current_elem_volume;
     // average deformation gradient
     ave_dfgrd_det /= _current_elem_volume;
-  }
-  for (_qp = 0; _qp < _qrule->n_points(); ++_qp)
-  {
-    if (_volumetric_locking_correction)
+
+    for (_qp = 0; _qp < _qrule->n_points(); ++_qp)
     {
       // Finalize volumetric locking correction
       _Fhat[_qp] *= std::cbrt(ave_Fhat.det() / _Fhat[_qp].det());
```

Comment:
You might want to pull out the `ave_Fhat.det()` in front of the loop and assign it to a `const Real`

Reply:
also below

Reply:
OK